<!doctype html>
<html class="no-js">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />
<link rel="index" title="Index" href="../genindex.html" /><link rel="search" title="Search" href="../search.html" /><link rel="next" title="bioimageloader.batch" href="batch.html" /><link rel="prev" title="bioimageloader.common" href="common.html" />

    <meta name="generator" content="sphinx-4.5.0, furo 2022.04.07"/>
        <title>bioimageloader.collections - bioimageloader</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo.css?digest=68f4518137b9aefe99b631505a2064c3c42c9852" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo-extensions.css?digest=30d1aed668e5c3a91c3e3bf6a60b675221979f0e" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../index.html"><div class="brand">bioimageloader</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon no-toc" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../index.html">
  
  
  <span class="sidebar-brand-text">bioimageloader</span>
  
</a><form class="sidebar-search-container" method="get" action="../search.html" role="search">
  <input class="sidebar-search" placeholder=Search name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation/index.html">Installation</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../user_guides/index.html">User Guides</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../user_guides/basic0_prepare_datasets.html">Basic: Prepare datasets</a></li>
<li class="toctree-l2"><a class="reference internal" href="../user_guides/basic1_basic.html">Basic: Basic usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="../user_guides/basic2_load_multi_dsets.html">Basic: Load multiple datasets</a></li>
<li class="toctree-l2"><a class="reference internal" href="../user_guides/basic3_deep_learning_with_pytorch.html">Basic: Deep learning with PyTorch</a></li>
<li class="toctree-l2"><a class="reference internal" href="../user_guides/more0_split.html">More: Split training/test set</a></li>
<li class="toctree-l2"><a class="reference internal" href="../user_guides/guide_more1_speedup.html">More: Speeding up loading by pre-parsing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../user_guides/more2_subclassing.html">More: Modifying existing collections</a></li>
<li class="toctree-l2"><a class="reference internal" href="../user_guides/more3_custom_dataset.html">More: Writing a custom Dataset</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../notebooks/index.html">Notebooks</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/pca.html">ML: Data statistics with PCA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/bioimage_io.html">DL: Deep learning with <em>BioImage.IO</em></a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/benchmark.html">DL: StarDist and Cellpose</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/train_models.html">DL: Training Stardist and Cellpose Models</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/train_models.html#(tmp)-Setup">(tmp) Setup</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/table.html">DL: Benchmark Table</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../catalogue/index.html">Collection Catalogue</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../tips/index.html">Miscellaneous</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" role="switch" type="checkbox"/><label for="toctree-checkbox-3"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../tips/tensorflow.html">TensorFlow installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="../tips/torch.html">PyTorch installation</a></li>
</ul>
</li>
<li class="toctree-l1 current has-children"><a class="reference internal" href="index.html">API</a><input checked="" class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" role="switch" type="checkbox"/><label for="toctree-checkbox-4"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bioimageloader.html">bioimageloader</a></li>
<li class="toctree-l2"><a class="reference internal" href="base.html">bioimageloader.base</a></li>
<li class="toctree-l2"><a class="reference internal" href="common.html">bioimageloader.common</a></li>
<li class="toctree-l2 current current-page"><a class="current reference internal" href="#">bioimageloader.collections</a></li>
<li class="toctree-l2"><a class="reference internal" href="batch.html">bioimageloader.batch</a></li>
<li class="toctree-l2"><a class="reference internal" href="utils.html">bioimageloader.utils</a></li>
<li class="toctree-l2"><a class="reference internal" href="transforms.html">bioimageloader.transforms</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../contributing/index.html">Contributing</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container"><div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon no-toc" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <section id="module-bioimageloader.collections">
<span id="bioimageloader-collections"></span><h1>bioimageloader.collections<a class="headerlink" href="#module-bioimageloader.collections" title="Permalink to this headline">#</a></h1>
<p>Collection of public bioimage datasets</p>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC002">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC002</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc002.html#BBBC002"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC002" title="Permalink to this definition">#</a></dt>
<dd><p>Drosophila Kc167 cells</p>
<p>There are 10 fields of view of each sample, for a total of 50 fields of
view. The images were acquired on a Zeiss Axiovert 200M microscope. The
images provided here are a single channel, DNA. The image size is 512 x 512
pixels. The images are provided as 8-bit TIFF files.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Cell count available</p></li>
<li><dl class="simple">
<dt>ImageJ RoI available for 3 tiles</dt><dd><ul>
<li><p>CPvalid1_48_40x_Tiles_p0151DAPI_ROIs.zip</p></li>
<li><p>CPvalid1_340_40x_Tiles_p1175DAPI_ROIs.zip</p></li>
<li><p>CPvalid1_nodsRNA_40x_Tiles_p0219DAPI_ROIs.zip</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r8135d40b2dbd-1"><span class="brackets">1</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC002">https://bbbc.broadinstitute.org/BBBC002</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC002.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc002.html#BBBC002.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC002.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC002.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC002.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC004">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC004</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc004.html#BBBC004"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC004" title="Permalink to this definition">#</a></dt>
<dd><p>Synthetic cells</p>
<p>Biological application</p>
<p>One of the principal challenges in counting or segmenting nuclei is dealing with clustered nuclei.
To help assess algorithms’ performance in this regard, this synthetic image set consists of five
subsets with increasing degree of clustering.</p>
<p>Images</p>
<p>Five subsets of 20 images each are provided. Each image contains 300 objects, but the objects overlap
and cluster with different probabilities in the five subsets. The images were generated with the SIMCEP
simulating platform for fluorescent cell population images (Lehmussola et al., IEEE T. Med. Imaging, 2007 and Lehmussola et al., P. IEEE, 2008).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>References</strong></dt><dd></dd>
<dt><strong>———-</strong></dt><dd></dd>
<dt><strong>.. [1] https://bbbc.broadinstitute.org/BBBC004</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC004.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc004.html#BBBC004.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC004.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC004.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc004.html#BBBC004.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC004.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC004.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC004.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC004.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC004.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC006">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC006</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'equal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">image_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('hoechst',</span> <span class="pre">'phalloidin')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">z_ind</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">16</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc006.html#BBBC006"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC006" title="Permalink to this definition">#</a></dt>
<dd><p>Human U2OS cells (out of focus)</p>
<p>Images were acquired from one 384-well microplate containing U2OS cells
stained with Hoechst 33342 markers (to label nuclei) were imaged with an
exposure of 15 and 1000 ms for Hoechst and phalloidin respectively, at 20x
magnification, 2x binning, and 2 sites per well. For each site, the optimal
focus was found using laser auto-focusing to find the well bottom. The
automated microscope was then programmed to collect a z-stack of 32 image
sets (z = 16 at the optimal focal plane, 15 images above the focal plane, 16
below) with 2 μm between slices. Each image is 696 x 520 pixels in 16-bit
TIF format, LZW compression. Each image filename includes either ‘w1’ to
denote Hoechst images or ‘w2’ to denote phalloidin images.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘equal’, ‘cv2’, Sequence[float]}, default: ‘equal’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>image_ch</strong><span class="classifier">{‘hoechst’, ‘phalloidin’}, default: (‘hoechst’, ‘phalloidin’)</span></dt><dd><p>Which channel(s) to load as image. Make sure to give it as a Sequence
when choose a single channel.</p>
</dd>
<dt><strong>z_ind</strong><span class="classifier">int, default: 16</span></dt><dd><p>Select one z stack. Default is 16, because 16 is the most in-focus.</p>
</dd>
<dt><strong>TIF format, LZW compression. Each image filename includes either ‘w1’ to</strong></dt><dd></dd>
<dt><strong>denote Hoechst images or ‘w2’ to denote phalloidin images.</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>z-stack, z=16 is in-focus ones, sites (s1, s2)</p></li>
<li><p>Instance segmented</p></li>
<li><p>384 wells, 2 sites per well; 384 * 2 = 768 images</p></li>
<li><p>2 channels, w1=Hoechst, w2=phalloidin</p></li>
<li><p>Two channels usually overlap and when overlapped, it’s hard to distinguish
two channels anymore.</p></li>
<li><p>Saved in UINT16, but UINT12 practically. Max value caps at 4095.</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="rb51ea772a266-1"><span class="brackets">1</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC006">https://bbbc.broadinstitute.org/BBBC006</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC006.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc006.html#BBBC006.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC006.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC006.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc006.html#BBBC006.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC006.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC006.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC006.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC006.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC006.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC007">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC007</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'equal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">image_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('DNA',</span> <span class="pre">'actin')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">anno_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('DNA',)</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc007.html#BBBC007"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC007" title="Permalink to this definition">#</a></dt>
<dd><p>Drosophila Kc167 cells</p>
<p>Outline annotation</p>
<p>Images were acquired using a motorized Zeiss Axioplan 2 and a Axiocam MRm
camera, and are provided courtesy of the laboratory of David Sabatini at the
Whitehead Institute for Biomedical Research. Each image is roughly 512 x 512
pixels, with cells roughly 25 pixels in dimeter, and 80 cells per image on
average. The two channels (DNA and actin) of each image are stored in
separate gray-scale 8-bit TIFF files.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘equal’, ‘cv2’, Sequence[float]}, default: ‘equal’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>image_ch</strong><span class="classifier">{‘DNA’, ‘actin’}, default: (‘DNA’, ‘actin’)</span></dt><dd><p>Which channel(s) to load as image. Make sure to give it as a Sequence
when choose a single channel. Name matches to <cite>anno_ch</cite>.</p>
</dd>
<dt><strong>anno_ch</strong><span class="classifier">{‘DNA’, ‘actin’}, default: (‘DNA’,)</span></dt><dd><p>Which channel(s) to load as annotation. Make sure to give it as a
Sequence when choose a single channel.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>[4, 5, 11, 14, 15] have 3 channels but they are just all gray scale
images. Extra work is required in get_image().</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r211ad2d1f79e-1"><span class="brackets">1</span></dt>
<dd><p>Jones et al., in the Proceedings of the ICCV Workshop on Computer
Vision for Biomedical Image Applications (CVBIA), 2005.</p>
</dd>
<dt class="label" id="r211ad2d1f79e-2"><span class="brackets">2</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC007">https://bbbc.broadinstitute.org/BBBC007</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC007.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc007.html#BBBC007.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC007.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC007.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc007.html#BBBC007.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC007.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC007.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC007.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC007.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC007.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC008">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC008</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'equal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">image_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('DNA',</span> <span class="pre">'actin')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">anno_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('DNA',)</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc008.html#BBBC008"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC008" title="Permalink to this definition">#</a></dt>
<dd><p>Human HT29 colon-cancer cells <a class="reference internal" href="#rc3e26290bef1-1" id="id5">[1]</a></p>
<p>F/B semantic segmentation</p>
<p>The image set consists of 12 images. The samples were stained with Hoechst
(channel 1), pH3 (channel 2), and phalloidin (channel 3). Hoechst labels
DNA, which is present in the nucleus. Phalloidin labels actin, which is
present in the cytoplasm. The last stain, pH3, indicates cells in mitosis;
whereas this was important for Moffat et al.’s screen, it is irrelevant for
segmentation and counting, so this channel is left out.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <cite>transforms</cite> is set. Define the total length of the dataset.
If it is set, it overrides <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘cv2’, ‘equal’, Sequence[float]}, default: ‘equal’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>image_ch</strong><span class="classifier">{‘DNA’, ‘actin’}, default: (‘DNA’, ‘actin’)</span></dt><dd><p>Which channel(s) to load as image. Make sure to give it as a Sequence
when choose a single channel.</p>
</dd>
<dt><strong>anno_ch</strong><span class="classifier">{‘DNA’, ‘actin’}, default: (‘DNA’,)</span></dt><dd><p>Which channel(s) to load as annotation. Make sure to give it as a
Sequence when choose a single channel.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Annotation F/B: BG=1, FG=0; very annoying…</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="rc3e26290bef1-1"><span class="brackets"><a class="fn-backref" href="#id5">1</a></span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC008">https://bbbc.broadinstitute.org/BBBC008</a></p>
</dd>
<dt class="label" id="rc3e26290bef1-2"><span class="brackets">2</span></dt>
<dd><p>Carpenter et al., Genome Biology, 2006</p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC008.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc008.html#BBBC008.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC008.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC008.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc008.html#BBBC008.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC008.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC008.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC008.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC008.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC008.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC009">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC009</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc009.html#BBBC009"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC009" title="Permalink to this definition">#</a></dt>
<dd><p>Human red blood cells</p>
<p>This image set consists of five differential interference contrast (DIC) images of red bood cells.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r616217d02254-1"><span class="brackets">1</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC009">https://bbbc.broadinstitute.org/BBBC009</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC009.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc009.html#BBBC009.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC009.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC009.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc009.html#BBBC009.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC009.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC009.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC009.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC009.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC009.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC013">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC013</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'equal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">image_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('GFP',</span> <span class="pre">'DNA')</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc013.html#BBBC013"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC013" title="Permalink to this definition">#</a></dt>
<dd><p>Human U2OS cells cytoplasm–nucleus translocation</p>
<p>The images were acquired at BioImage on the IN Cell Analyzer 3000 using the
Trafficking Data Analysis Module, with one image per channel (Channel 1 =
FKHR-GFP; Channel 2 = DNA). Image size is 640 x 640 pixels. Images are
available in native FRM format or 8-bit BMP.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘equal’, ‘cv2’, Sequence[float]}, default: ‘equal’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>image_ch</strong><span class="classifier">{‘GFP’, ‘DNA’}, default: (‘GFP’, ‘DNA’)</span></dt><dd><p>Which channel(s) to load as image. Make sure to give it as a Sequence
when choose a single channel.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Two formats are available; FRM and BMP</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r162ffbf2d2d3-1"><span class="brackets">1</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC013">https://bbbc.broadinstitute.org/BBBC013</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC013.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc013.html#BBBC013.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC013.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC013.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC013.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC014">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC014</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'equal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">image_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('DAPI',</span> <span class="pre">'FITC')</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc014.html#BBBC014"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC014" title="Permalink to this definition">#</a></dt>
<dd><p>Human U2OS cells cytoplasm–nucleus translocation</p>
<p>This 96-well plate has images of cytoplasm to nucleus translocation of the
transcription factor NFκB in MCF7 (human breast adenocarcinoma cell line)
and A549 (human alveolar basal epithelial) cells in response to TNFα
concentration.</p>
<p>Images are at 10x objective magnification. The plate was acquired at Vitra
Bioscience on the CellCard reader. For each well there is one field with two
images: a nuclear counterstain (DAPI) image and a signal stain (FITC) image.
Image size is 1360 x 1024 pixels. Images are in 8-bit BMP format.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘equal’, ‘cv2’, Sequence[float]}, default: ‘equal’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>image_ch</strong><span class="classifier">{‘DAPI’, ‘FITC’}, default: (‘DAPI’, ‘FITC’)</span></dt><dd><p>Which channel(s) to load as image. Make sure to give it as a Sequence
when choose a single channel.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Second channel is usually very clear with a few artifacts</p></li>
<li><p>Biological annotation</p></li>
<li><p>CellProfiler’s LoadText module format annotation also available (not
implemented)</p></li>
<li><p>Zoom in?</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="rfcd7ae1a10e8-1"><span class="brackets">1</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC014">https://bbbc.broadinstitute.org/BBBC014</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC014.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc014.html#BBBC014.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC014.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC014.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC014.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC015">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC015</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'equal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">image_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('b2AR',</span> <span class="pre">'arrestin')</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc015.html#BBBC015"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC015" title="Permalink to this definition">#</a></dt>
<dd><p>Human U2OS cells transfluor</p>
<p>The images are of a human osteosarcoma cell line (U2OS) co-expressing beta2
(b2AR) adrenergic receptor and arrestin-GFP protein molecules. The receptor
was modified-type that generates “vesicle-type” spots upon ligand
stimulation.</p>
<p>The plate was acquired on iCyte imaging cytometer with iCyte software
version 2.5.1. Image file format is JPEG with one image for green channel
and one image for crimson channel. Image size is 1000 x 768 pixels.</p>
<p>This image set has a portion of a 96-well plate containing 3 replica rows
and 12 concentration points of isoproterenol. In each well four fields were
acquired. File name structure: &lt;well-number&gt;_&lt;field&gt;_&lt;channel&gt;.JPG</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘cv2’, ‘equal’, Sequence[float]}, default: ‘equal’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>image_ch</strong><span class="classifier">{‘b2AR’, ‘arrestin’}, default: (‘b2AR’, ‘arrestin’)</span></dt><dd><p>Which channel(s) to load as image. Make sure to give it as a Sequence
when choose a single channel.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>2 channels (Green, Crimson?), texture in green channel</p></li>
<li><p>Crimson channel…?</p></li>
<li><p>RGB channel is all the same in each image file</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r0a2db2aadddd-1"><span class="brackets">1</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC015">https://bbbc.broadinstitute.org/BBBC015</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC015.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc015.html#BBBC015.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC015.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC015.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC015.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC016">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC016</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'equal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">image_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('GFP',</span> <span class="pre">'DNA')</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc016.html#BBBC016"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC016" title="Permalink to this definition">#</a></dt>
<dd><p>Human U2OS cells transfluor</p>
<p>This image set is of a Transfluor assay where an orphan GPCR is stably
integrated into the b-arrestin GFP expressing U2OS cell line. After one hour
incubation with a compound the cells were fixed with (formaldehyde).</p>
<p>The plate was read on Cellomics ArrayScan HCS Reader using the GPCR
Bioapplication. File format is 8-bit TIFF with one image for green channel
(GFP) and one image for blue channel (DNA). Image size is 512 x 512 pixels.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘equal’, ‘cv2’, Sequence[float]}, default: ‘equal’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>image_ch</strong><span class="classifier">{‘GFP’, ‘DNA’}, default: (‘GFP’, ‘DNA’)</span></dt><dd><p>Which channel(s) to load as image. Make sure to give it as a Sequence
when choose a single channel.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>2 channels (G,B), nuclei are Blue</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r24fe942e420a-1"><span class="brackets">1</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC016">https://bbbc.broadinstitute.org/BBBC016</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC016.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc016.html#BBBC016.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC016.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC016.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC016.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC018">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC018</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'equal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">image_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('DNA',</span> <span class="pre">'actin',</span> <span class="pre">'pH3')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">anno_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('DNA',)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_missing_pairs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc018.html#BBBC018"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC018" title="Permalink to this definition">#</a></dt>
<dd><p>Human HT29 colon-cancer cells (diverse phenotypes)</p>
<p>The image set consists of 56 fields of view (4 from each of 14 samples).
Because there are three channels, there are 168 image files. (The samples
were stained with Hoechst 33342, pH3, and phalloidin. Hoechst 33342 is a DNA
stain that labels the nucleus. Phospho-histone H3 indicates mitosis.
Phalloidin labels actin, which is present in the cytoplasm.) The samples are
the top-scoring sample from each of Jones et al.’s classifiers, as listed in
the file SamplesScores.zip in their supplement. The files are in DIB format,
as produced by the Cellomics ArrayScan instrument at the Whitehead–MIT
Bioimaging Center. We recommend using Bio-Formats to read the DIB files.
Each image is 512 x 512 pixels.</p>
<p>The filenames are of the form wellidx-channel.DIB, where wellidx is the
five-digit well index (from Jones et al.’s supplement) and channel is either
DNA, actin, or pH3, depending on the channel.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘equal’, ‘cv2’, Sequence[float]}, default: ‘equal’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>image_ch</strong><span class="classifier">{‘DNA’, ‘actin’, ‘pH3’}, default: (‘DNA’, ‘actin’, ‘pH3’)</span></dt><dd><p>Which channel(s) to load as image. Make sure to give it as a Sequence
when choose a single channel.</p>
</dd>
<dt><strong>anno_ch</strong><span class="classifier">{‘DNA’, ‘actin’}, default: (‘DNA’,)</span></dt><dd><p>Which channel(s) to load as annotation. Make sure to give it as a
Sequence when choose a single channel.</p>
</dd>
<dt><strong>drop_missing_pairs</strong><span class="classifier">bool, default: True</span></dt><dd><p>Valid only if <cite>output=’both’</cite>. It will drop images that do not have mask
pairs.</p>
</dd>
</dl>
</dd>
<dt class="field-even">Other Parameters</dt>
<dd class="field-even"><dl class="simple">
<dt><strong>image_ch</strong><span class="classifier">{‘DNA’, ‘actin’}, default: (‘DNA’, ‘actin’)</span></dt><dd><p>Which channel(s) to load as image. Make sure to give it as a Sequence
when choose a single channel.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>BBBC018_v1_images/10779 annotation is missing. len(anno_dict) =
len(file_list) - 1; ind={26}</p>
<blockquote>
<div><ul class="simple">
<li><p>PosixPath(‘BBBC018_v1_images/10779-DNA.DIB’)</p></li>
<li><p>PosixPath(‘BBBC018_v1_images/10779-actin.DIB’)</p></li>
<li><p>PosixPath(‘BBBC018_v1_images/10779-pH3.DIB’)</p></li>
</ul>
</div></blockquote>
<p>This one is not properly saved after annotation. It has annotation overlaid
on top to image. Need to filter <a href="#id13"><span class="problematic" id="id14">``</span></a>mask==255`.</p>
<blockquote>
<div><ul class="simple">
<li><p>‘BBBC018_v1_outlines/17675-nuclei.png’</p></li>
</ul>
</div></blockquote>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Every DIB has 3 channels (Order = (DNA,actin,pH3)). The second one is the
object.</p></li>
<li><p>DNA -&gt; Nuceli</p></li>
<li><p>Actin -&gt; Cell</p></li>
<li><p>Annotation is outline one, but every anno is closed so binary_fill_holes
works fine</p></li>
<li><p>For some reason annotation is y inverted</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r02c115b38c80-1"><span class="brackets">1</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC018">https://bbbc.broadinstitute.org/BBBC018</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC018.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc018.html#BBBC018.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC018.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC018.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc018.html#BBBC018.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC018.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC018.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC018.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC018.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC018.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC020">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC020</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'equal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">image_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('nuclei',</span> <span class="pre">'cells')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">anno_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('nuclei',)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_missing_pairs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc020.html#BBBC020"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC020" title="Permalink to this definition">#</a></dt>
<dd><p>Murine bone-marrow derived macrophages</p>
<p>The image set consists of 25 images, each consisting of three channels. The
samples were stained with DAPI and CD11b/APC. In addition to this, a merged
image is provided. DAPI labels the nuclei and CD11b/APC the cell surface.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘equal’, ‘cv2’, Sequence[float]}, default: ‘equal’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>image_ch</strong><span class="classifier">{‘cell’, ‘nuclei’}, default: (‘cell’, ‘nuclei’)</span></dt><dd><p>Which channel(s) to load as image. Make sure to give it as a Sequence
when choose a single channel.</p>
</dd>
<dt><strong>anno_ch</strong><span class="classifier">{‘nuclei’, ‘cells’}, default: (‘nuclei’,)</span></dt><dd><p>Which channel(s) to load as annotation. Make sure to give it as a
Sequence when choose a single channel.</p>
</dd>
<dt><strong>drop_missing_pairs</strong><span class="classifier">bool, default: True</span></dt><dd><p>Valid only if <cite>output=’both’</cite>. It will drop images that do not have mask
pairs.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>5 annotations are missing: ind={17,18,19,20,21}
[jw-30min 1, jw-30min 2, jw-30min 3, jw-30min 4, jw-30min 5]</p>
<blockquote>
<div><ul class="simple">
<li><p>./BBBC020_v1_images/jw-30min 1/jw-30min 1_(c1+c5).TIF</p></li>
<li><p>./BBBC020_v1_images/jw-30min 2/jw-30min 2_(c1+c5).TIF</p></li>
<li><p>./BBBC020_v1_images/jw-30min 3/jw-30min 3_(c1+c5).TIF</p></li>
<li><p>./BBBC020_v1_images/jw-30min 4/jw-30min 4_(c1+c5).TIF</p></li>
<li><p>./BBBC020_v1_images/jw-30min 5/jw-30min 5_(c1+c5).TIF</p></li>
</ul>
</div></blockquote>
<ul class="simple">
<li><p>BBC020_v1_outlines_nuclei/jw-15min 5_c5_43.TIF exists but corrupted</p></li>
</ul>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Anotations are instance segmented where each of them is saved as a single
image file. It loads and aggregates them as a single array. Label loaded
after will override the one loaded before. If you do not want this
behavior, make a subclass out of this class and override <code class="docutils literal notranslate"><span class="pre">get_mask()</span></code>
method, accordingly.</p></li>
<li><dl class="simple">
<dt>2 channels; R channel is the same as G, R==G!=B</dt><dd><p>Assign 0 to red channel</p>
</dd>
</dl>
</li>
<li><p>BBBC has received a complaint that “BBB020_v1_outlines_nuclei” appears
incomplete and we have been unable to obtain the missing images from the
original contributor.</p></li>
<li><p>Nuclei anno looks good</p></li>
<li><p>Should separte nuclei and cells annotation; if <code class="docutils literal notranslate"><span class="pre">anno_ch=None</span></code>,
<code class="docutils literal notranslate"><span class="pre">anno_dict</span></code> becomes a mess.</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r25dcadb594d9-1"><span class="brackets">1</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC020">https://bbbc.broadinstitute.org/BBBC020</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC020.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc020.html#BBBC020.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC020.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC020.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">lst_p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc020.html#BBBC020.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC020.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC020.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC020.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC020.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC020.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC021">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC021</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'equal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">image_ch</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('DNA',</span> <span class="pre">'actin',</span> <span class="pre">'tublin')</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc021.html#BBBC021"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC021" title="Permalink to this definition">#</a></dt>
<dd><p>Human MCF7 cells – compound-profiling experiment <a class="reference internal" href="#rd795ca58f219-1" id="id17">[1]</a></p>
<p>The images are of MCF-7 breast cancer cells treated for 24 h with a
collection of 113 small molecules at eight concentrations. The cells were
fixed, labeled for DNA, F-actin, and Β-tubulin, and imaged by fluorescent
microscopy as described [Caie et al. Molecular Cancer Therapeutics, 2010].</p>
<p>There are 39,600 image files (13,200 fields of view imaged in three
channels) in TIFF format. We provide the images in 55 ZIP archives, one for
each microtiter plate. The archives are ~750 MB each.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘equal’, ‘cv2’, Sequence[float]}, default: ‘equal’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>image_ch</strong><span class="classifier">{‘DNA’, ‘actin’, ‘tublin’}, default: (‘DNA’, ‘actin’, ‘tublin’)</span></dt><dd><p>Which channel(s) to load as image. Make sure to give it as a Sequence
when choose a single channel.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>HUGE dataset</p></li>
<li><dl class="simple">
<dt>3 channels</dt><dd><ul>
<li><p>w1 (DNA) -&gt; Blue</p></li>
<li><p>w2 (actin?) -&gt; Green</p></li>
<li><p>w4 (tublin??)-&gt; Red</p></li>
</ul>
</dd>
</dl>
</li>
<li><p>UINT16</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="rd795ca58f219-1"><span class="brackets"><a class="fn-backref" href="#id17">1</a></span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC021">https://bbbc.broadinstitute.org/BBBC021</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC021.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc021.html#BBBC021.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC021.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC021.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC021.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC026">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC026</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc026.html#BBBC026"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC026" title="Permalink to this definition">#</a></dt>
<dd><p>Human Hepatocyte and Murine Fibroblast cells – Co-culture experiment</p>
<p>This 384-well plate has images of co-cultured hepatocytes and fibroblasts.
Every other well is populated (A01, A03, …, C01, C03, …) such that 96
wells comprise the data. Each well has 9 sites and thus 9 images associated,
totaling 864 images.</p>
<p>For each well there is one field and a single image nuclear image (Hoecsht).
Images are in 8-bit PNG format.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Only centers are annotated for 5 imgages (not implemented)</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r9bed8913f06d-1"><span class="brackets">1</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC026">https://bbbc.broadinstitute.org/BBBC026</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC026.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc026.html#BBBC026.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC026.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC026.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC026.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC030">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC030</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc030.html#BBBC030"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC030" title="Permalink to this definition">#</a></dt>
<dd><p>Chinese Hamster Ovary Cells</p>
<p>The image set consists of 60 Differential Interference Contrast (DIC) images of Chinese Hamster Ovary (CHO) cells.
The images are taken on an Olympus Cell-R microscope with a 20x lens at the time when the cell initiated their attachment to the bottom of the dish.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>See Also</strong></dt><dd></dd>
<dt><strong>——–</strong></dt><dd></dd>
<dt><strong>MaskDataset</strong><span class="classifier">Super class</span></dt><dd></dd>
<dt><strong>DatasetInterface</strong><span class="classifier">Interface</span></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r98976f45e7c2-1"><span class="brackets">1</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC030">https://bbbc.broadinstitute.org/BBBC030</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC030.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc030.html#BBBC030.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC030.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC030.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc030.html#BBBC030.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC030.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC030.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC030.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC030.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC030.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC039">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC039</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">training</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc039.html#BBBC039"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC039" title="Permalink to this definition">#</a></dt>
<dd><p>Nuclei of U2OS cells in a chemical screen <a class="reference internal" href="#r584a9f182b09-1" id="id21">[1]</a></p>
<p>This data set has a total of 200 fields of view of nuclei captured with
fluorescence microscopy using the Hoechst stain. These images are a sample
of the larger BBBC022 chemical screen. The images are stored as TIFF files
with 520x696 pixels at 16 bits.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>training</strong><span class="classifier">bool, default: True</span></dt><dd><p>Load training set if True, else load testing one</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><dl class="simple">
<dt>Split (training/valiadation/test)</dt><dd><ul>
<li><p><cite>training=True</cite> combines ‘training’ with ‘validation’</p></li>
</ul>
</dd>
</dl>
</li>
<li><p>Annotate objs not touching each other with 1 and use 2, 3, … for the
touching ones. It is great and clever, but it does not follow the form of
other instance segmented masks. <code class="docutils literal notranslate"><span class="pre">get_mask()</span></code> will make a instance
labeled mask (each obj has unique labels). After labeling max label is 231
for training, and 202 for test. So having masks of dtype UINT8 is fine.</p></li>
<li><p>Max label is 3 (in original annotation)</p></li>
<li><p>Sample of larger BBBC022 and did manual segmentation</p></li>
<li><p>Possible overlap some with DSB2018</p></li>
<li><p>Mask is png but (instance) value is only stored in RED channel</p></li>
<li><p>Maximum value is 2**12</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r584a9f182b09-1"><span class="brackets"><a class="fn-backref" href="#id21">1</a></span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC039">https://bbbc.broadinstitute.org/BBBC039</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC039.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc039.html#BBBC039.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC039.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC039.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc039.html#BBBC039.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC039.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC039.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC039.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC039.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC039.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC041">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">BBBC041</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'cv2'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">training</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc041.html#BBBC041"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC041" title="Permalink to this definition">#</a></dt>
<dd><ol class="upperalpha simple" start="16">
<li><p>vivax (malaria) infected human blood smears <a class="reference internal" href="#rb7f2a6902816-1" id="id23">[1]</a></p></li>
</ol>
<p>Images are in .png or .jpg format. There are 3 sets of images consisting of
1364 images (~80,000 cells) with different researchers having prepared each
one: from Brazil (Stefanie Lopes), from Southeast Asia (Benoit Malleret),
and time course (Gabriel Rangel). Blood smears were stained with Giemsa
reagent.</p>
<p>These images were contributed by Jane Hung of MIT and the Broad Institute in
Cambridge, MA. <a class="reference internal" href="#rb7f2a6902816-1" id="id24">[1]</a></p>
<p>There is also a Github reposity that lists malaria parasite imaging datasets
(blood smears) <a class="reference internal" href="#rb7f2a6902816-2" id="id25">[2]</a>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">`transforms`</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘cv2’, ‘equal’, Sequence[float]}, default: ‘cv2’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>training</strong><span class="classifier">bool, default: True</span></dt><dd><p>Load training set if True, else load testing one</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Label categories: all 7 cats, [‘difficult’, ‘gametocyte’, ‘leukocyte’,
‘red blood cell’, ‘ring’, ‘schizont’, ‘trophozoite’]</p></li>
<li><p>1208/120 training/test split. So not 1368 images as written in the
description.</p></li>
<li><p>png and jpg extension; training images are in RGB space in PNG format,
while test images are in YUV space in JPEG format.</p></li>
<li><p>YUV will be automatically detected when read and cast to RGB</p></li>
<li><p>Two resolutions; depending on training/test: (1200, 1600) for training,
(1383, 1944) for test</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="rb7f2a6902816-1"><span class="brackets">1</span><span class="fn-backref">(<a href="#id23">1</a>,<a href="#id24">2</a>)</span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC041">https://bbbc.broadinstitute.org/BBBC041</a></p>
</dd>
<dt class="label" id="rb7f2a6902816-2"><span class="brackets"><a class="fn-backref" href="#id25">2</a></span></dt>
<dd><p><a class="reference external" href="https://github.com/tobsecret/Awesome_Malaria_Parasite_Imaging_Datasets">https://github.com/tobsecret/Awesome_Malaria_Parasite_Imaging_Datasets</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC041.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_bbbc041.html#BBBC041.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.BBBC041.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.BBBC041.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.BBBC041.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.Cellpose">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">Cellpose</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'equal'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">training</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gray_is_not_green</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">specialized_data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_cellpose.html#Cellpose"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.Cellpose" title="Permalink to this definition">#</a></dt>
<dd><p>Dataset for Cellpose <a class="reference internal" href="#r526138c83466-1" id="id28">[1]</a>, <a class="reference internal" href="#r526138c83466-2" id="id29">[2]</a></p>
<p>Cellpose: a generalist algorithm for cellular segmentation</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘cv2’, ‘equal’, Sequence[float]}, default: ‘cv2’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>training</strong><span class="classifier">bool, default: True</span></dt><dd><p>Load training set if True, else load testing one.</p>
</dd>
<dt><strong>gray_is_not_green</strong><span class="classifier">bool, default: True</span></dt><dd><p>Proper grayscale. Green channel value will be broadcast to all channels.</p>
</dd>
<dt><strong>specialized_data</strong><span class="classifier">bool, default: False</span></dt><dd><p>Load “specialized data” mentioned in the paper <a class="reference internal" href="#r526138c83466-1" id="id30">[1]</a>.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Download link is hard to find <a class="reference internal" href="#r526138c83466-3" id="id31">[3]</a></p></li>
<li><p>It is a complete dataset by itself, meaning that it is not intended to be
mixed or concatenated with others. It consists of various sources of
images, not only bioimages but also images of fruits, rocks and etc.</p></li>
<li><p>All images have 3 channels, but technically they are not RGB. Every images
have values on the second channel and if there is more signal, then it
goes to the first one. There is no image that has values on the last
channel. As a result, when visualized in RGB, they look all green and red.
In particular, for this reason, grayscale images have signal in the second
channel and look green. <code class="docutils literal notranslate"><span class="pre">gray_is_not_green</span></code> argument address that.</p></li>
<li><p>Built-in grayscale conversion methods are not correct for this dataset.
The conversion should be channel-agnostic.</p></li>
<li><p>Currently, <code class="docutils literal notranslate"><span class="pre">gray_is_not_green=False</span></code> and <code class="docutils literal notranslate"><span class="pre">grayscale=True</span></code> will reduce
values of single channel images 1/3 times.</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r526138c83466-1"><span class="brackets">1</span><span class="fn-backref">(<a href="#id28">1</a>,<a href="#id30">2</a>)</span></dt>
<dd><p>C. Stringer, M. Michaelos, and M. Pachitariu, “Cellpose: a generalist
algorithm for cellular segmentation,” bioRxiv, p. 2020.02.02.931238, Feb.
2020, doi: 10.1101/2020.02.02.931238.</p>
</dd>
<dt class="label" id="r526138c83466-2"><span class="brackets"><a class="fn-backref" href="#id29">2</a></span></dt>
<dd><p><a class="reference external" href="https://github.com/mouseLand/cellpose">https://github.com/mouseLand/cellpose</a></p>
</dd>
<dt class="label" id="r526138c83466-3"><span class="brackets"><a class="fn-backref" href="#id31">3</a></span></dt>
<dd><p><a class="reference external" href="https://www.cellpose.org/dataset">https://www.cellpose.org/dataset</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.Cellpose.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_cellpose.html#Cellpose.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.Cellpose.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.Cellpose.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_cellpose.html#Cellpose.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.Cellpose.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.Cellpose.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.Cellpose.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.Cellpose.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.Cellpose.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.ComputationalPathology">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">ComputationalPathology</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'cv2'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mask_tif</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_compath.html#ComputationalPathology"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.ComputationalPathology" title="Permalink to this definition">#</a></dt>
<dd><p>A Dataset and a Technique for Generalized Nuclear Segmentation for
Computational Pathology <a class="reference internal" href="#rd333aeb16733-1" id="id35">[1]</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">`transforms`</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘cv2’, ‘equal’, Sequence[float]}, default: ‘cv2’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>mask_tif</strong><span class="classifier">bool, default: False</span></dt><dd><p>Instead of parsing every xml file to reconstruct mask image arrays, use
pre-drawn mask tif files which should reside in the same folder as
annotation xml files.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Resolution of all images is (1000,1000)</p></li>
<li><p>gt is converted from annotation recorded in xml format</p></li>
<li><p>gt has dtype of torch.float64, converted from numpy.uint16, and it has
value ‘num_objects’ * 255 because it is base-transformed</p></li>
<li><p>The origianl dataset provides annotation in xml format, which takes
long time to parse and to reconstruct mask images dynamically during
training. Drawing masks beforehand makes training much faster. Use
<code class="docutils literal notranslate"><span class="pre">mask_tif</span></code> in that case.</p></li>
<li><p>When <code class="docutils literal notranslate"><span class="pre">augmenters</span></code> is provided, set the <code class="docutils literal notranslate"><span class="pre">num_samples</span></code> argument
30x1000x1000 -&gt; 16x30=480 patches. Thus, the default <code class="docutils literal notranslate"><span class="pre">num_samples=720</span></code>
(x1.5)</p></li>
<li><p>dtype of ‘gt’ is int16. However, to make batching easier, it will be
casted to float32</p></li>
<li><p>Be careful about types of augmenters; avoid interpolation</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="rd333aeb16733-1"><span class="brackets"><a class="fn-backref" href="#id35">1</a></span></dt>
<dd><p>N. Kumar, R. Verma, S. Sharma, S. Bhargava, A. Vahadane, and A.
Sethi, “A Dataset and a Technique for Generalized Nuclear Segmentation
for Computational Pathology,” IEEE Transactions on Medical Imaging, vol.
36, no. 7, pp. 1550–1560, Jul. 2017, doi: 10.1109/TMI.2017.2677499.</p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.ComputationalPathology.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_compath.html#ComputationalPathology.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.ComputationalPathology.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.ComputationalPathology.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_compath.html#ComputationalPathology.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.ComputationalPathology.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.ComputationalPathology.save_xml_to_tif">
<span class="sig-name descname"><span class="pre">save_xml_to_tif</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_compath.html#ComputationalPathology.save_xml_to_tif"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.ComputationalPathology.save_xml_to_tif" title="Permalink to this definition">#</a></dt>
<dd><p>Parse .xml to mask and write it as tiff file</p>
<p>Having masks in images is much faster than parsing .xml for each call.
This func iterates through <code class="docutils literal notranslate"><span class="pre">anno_dict</span></code>, parse and save each in .tif
format in the same annotation directory. Re-initiate an instance with
<code class="docutils literal notranslate"><span class="pre">mask_tif</span></code> argument to load them.</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.ComputationalPathology.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">list</span></em><a class="headerlink" href="#bioimageloader.collections.ComputationalPathology.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.ComputationalPathology.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.ComputationalPathology.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.DSB2018">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">DSB2018</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'cv2'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">training</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_dsb2018.html#DSB2018"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.DSB2018" title="Permalink to this definition">#</a></dt>
<dd><p>Data Science Bowl 2018 <a class="reference internal" href="#r6ad5ab3120e1-1" id="id37">[1]</a> also known as BBBC038 <a class="reference internal" href="#r6ad5ab3120e1-2" id="id38">[2]</a></p>
<p>Find the nuclei in divergent images to advance medical discovery</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘cv2’, ‘equal’, Sequence[float]}, default: ‘cv2’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>training</strong><span class="classifier">bool, default: True</span></dt><dd><p>Load training set if True, else load testing one</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r6ad5ab3120e1-1"><span class="brackets"><a class="fn-backref" href="#id37">1</a></span></dt>
<dd><p><a class="reference external" href="https://www.kaggle.com/c/data-science-bowl-2018/">https://www.kaggle.com/c/data-science-bowl-2018/</a></p>
</dd>
<dt class="label" id="r6ad5ab3120e1-2"><span class="brackets"><a class="fn-backref" href="#id38">2</a></span></dt>
<dd><p><a class="reference external" href="https://bbbc.broadinstitute.org/BBBC038/">https://bbbc.broadinstitute.org/BBBC038/</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.DSB2018.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_dsb2018.html#DSB2018.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.DSB2018.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.DSB2018.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">anno</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Any</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_dsb2018.html#DSB2018.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.DSB2018.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.DSB2018.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.DSB2018.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.DSB2018.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">dict</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.DSB2018.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.DigitalPathology">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">DigitalPathology</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'cv2'</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_digitpath.html#DigitalPathology"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.DigitalPathology" title="Permalink to this definition">#</a></dt>
<dd><p>Deep learning for digital pathology image analysis: A comprehensive
tutorial with selected use cases <a class="reference internal" href="#rdcfe2d43dee5-1" id="id41">[1]</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘cv2’, ‘equal’, Sequence[float]}, default: ‘cv2’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along channel
axis, then divides it by the number of expected channels.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Annotation is partial</p></li>
<li><p>Boolean mask to UINT8 mask (0, 255)</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="rdcfe2d43dee5-1"><span class="brackets"><a class="fn-backref" href="#id41">1</a></span></dt>
<dd><p>A. Janowczyk and A. Madabhushi, “Deep learning for digital pathology
image analysis: A comprehensive tutorial with selected use cases,” J
Pathol Inform, vol. 7, Jul. 2016, doi: 10.4103/2153-3539.186902.</p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.DigitalPathology.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_digitpath.html#DigitalPathology.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.DigitalPathology.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.DigitalPathology.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_digitpath.html#DigitalPathology.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.DigitalPathology.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.DigitalPathology.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.DigitalPathology.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.DigitalPathology.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.DigitalPathology.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.FRUNet">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">FRUNet</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">normalize</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_frunet.html#FRUNet"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.FRUNet" title="Permalink to this definition">#</a></dt>
<dd><p>FRU-Net: Robust Segmentation of Small Extracellular Vesicles <a class="reference internal" href="#r4af27370572d-1" id="id43">[1]</a></p>
<p>TEM images</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>normalize</strong><span class="classifier">bool, default: True</span></dt><dd><p>Normalize each image by its maximum value and cast it to UINT8.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Originally, dtype is UINT16</p></li>
<li><p>Max value is 20444, but contrast varies a lot. For example, some images
have value less than 0.05 of 2^16, which makes images not visible.
Normalization may be needed. Init param <code class="docutils literal notranslate"><span class="pre">normalize</span></code> is set to True by
default for this reason. For each image, it calculates maximum value and
divide by it.</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r4af27370572d-1"><span class="brackets"><a class="fn-backref" href="#id43">1</a></span></dt>
<dd><p>E. Gómez-de-Mariscal, M. Maška, A. Kotrbová, V. Pospíchalová, P.
Matula, and A. Muñoz-Barrutia, “Deep-Learning-Based Segmentation of Small
Extracellular Vesicles in Transmission Electron Microscopy Images,”
Scientific Reports, vol. 9, no. 1, Art. no. 1, Sep. 2019, doi:
10.1038/s41598-019-49431-3.</p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.FRUNet.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_frunet.html#FRUNet.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.FRUNet.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.FRUNet.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_frunet.html#FRUNet.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.FRUNet.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.FRUNet.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.FRUNet.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.FRUNet.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.FRUNet.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.LIVECell">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">LIVECell</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">training</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mask_tif</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_livecell.html#LIVECell"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.LIVECell" title="Permalink to this definition">#</a></dt>
<dd><p>LIVECEll: A large-scale dataset for label-free live cell segmentation
<a class="reference internal" href="#rb57a48c69b8f-1" id="id45">[1]</a></p>
<p>“LIVECell - A large-scale dataset for label-free live cell segmentation” by
Edlund et. al. 2021 <a class="reference internal" href="#rb57a48c69b8f-2" id="id46">[2]</a></p>
<p>Light microscopy is a cheap, accessible, non-invasive modality that when
combined with well-established protocols of two-dimensional cell culture
facilitates high-throughput quantitative imaging to study biological
phenomena. Accurate segmentation of individual cells enables exploration of
complex biological questions, but this requires sophisticated imaging
processing pipelines due to the low contrast and high object density. Deep
learning-based methods are considered state-of-the-art for most computer
vision problems but require vast amounts of annotated data, for which there
is no suitable resource available in the field of label-free cellular
imaging. To address this gap we present LIVECell, a high-quality, manually
annotated and expert-validated dataset that is the largest of its kind to
date, consisting of over 1.6 million cells from a diverse set of cell
morphologies and culture densities. To further demonstrate its utility, we
provide convolutional neural network-based models trained and evaluated on
LIVECell.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>training</strong><span class="classifier">bool, default: True</span></dt><dd><p>Load training set if True, else load testing one</p>
</dd>
<dt><strong>mask_tif</strong><span class="classifier">bool, default: False</span></dt><dd><p>Use saved COCO annotations as tif mask images in a new ./root_dir/masks
directory. It will greatly improve loading speed. Available after
calling <code class="docutils literal notranslate"><span class="pre">save_coco_to_tif()</span></code>.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Annotation in MS COCO format <a class="reference internal" href="#rb57a48c69b8f-3" id="id47">[3]</a>. Parsing it takes time`.</p></li>
<li><p>Currently not supporting dynamically parsing COCO annotation due to slow
speed. Pre-parse masks in .tif format by calling <code class="docutils literal notranslate"><span class="pre">save_coco_to_tif()</span></code>.</p></li>
<li><p>Validation set is originally separted from training set. Currently they
are combined <code class="docutils literal notranslate"><span class="pre">training=True</span></code>.</p></li>
<li><p>Single cells subsets are not covered</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="rb57a48c69b8f-1"><span class="brackets"><a class="fn-backref" href="#id45">1</a></span></dt>
<dd><p><a class="reference external" href="https://sartorius-research.github.io/LIVECell/">https://sartorius-research.github.io/LIVECell/</a></p>
</dd>
<dt class="label" id="rb57a48c69b8f-2"><span class="brackets"><a class="fn-backref" href="#id46">2</a></span></dt>
<dd><p><a class="reference external" href="https://www.nature.com/articles/s41592-021-01249-6">https://www.nature.com/articles/s41592-021-01249-6</a></p>
</dd>
<dt class="label" id="rb57a48c69b8f-3"><span class="brackets"><a class="fn-backref" href="#id47">3</a></span></dt>
<dd><p><a class="reference external" href="https://cocodataset.org/">https://cocodataset.org/</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.LIVECell.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_livecell.html#LIVECell.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.LIVECell.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.LIVECell.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_livecell.html#LIVECell.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.LIVECell.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.LIVECell.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.LIVECell.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.LIVECell.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.LIVECell.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.LIVECell.save_coco_to_tif">
<span class="sig-name descname"><span class="pre">save_coco_to_tif</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_livecell.html#LIVECell.save_coco_to_tif"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.LIVECell.save_coco_to_tif" title="Permalink to this definition">#</a></dt>
<dd><p>Save the masks as tif files</p>
<p>Read training/val or test annotations from json. Make tif files under
‘masks/livecell_train_val_masks’ and ‘masks/livecell_test_masks’.</p>
<p>Initialize a new instance with setting <code class="docutils literal notranslate"><span class="pre">mask_tif=True</span></code> to load saved
masks.</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.MurphyLab">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">MurphyLab</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_missing_pairs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_broken_files</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filled_mask</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_murphylab.html#MurphyLab"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.MurphyLab" title="Permalink to this definition">#</a></dt>
<dd><p>Nuclei Segmentation In Microscope Cell Images: A Hand-Segmented Dataset
And Comparison Of Algorithms <a class="reference internal" href="#r392840f6ea36-1" id="id51">[1]</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str or pathlib.Path</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines
augmentation in sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>drop_missing_pairs</strong><span class="classifier">bool, default: True</span></dt><dd><p>Valid only if <cite>output=’both’</cite>. It will drop images that do not have
mask pairs.</p>
</dd>
<dt><strong>drop_broken_files</strong><span class="classifier">bool, default: True</span></dt><dd><p>Drop broken files that cannot be read</p>
</dd>
<dt><strong>filled_mask</strong><span class="classifier">bool, default: False</span></dt><dd><p>Use saved filled masks through <cite>fill_save_mask()</cite> method instead of
default boundary masks. If one would want to use manually modified
masks, the annotation files should have the same name as ‘<a href="#id52"><span class="problematic" id="id53">*</span></a>.xcf’
with modified suffix by ‘.png’.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>This dataset has many issues whose details can be found below. The simpleset
way is to drop those that cause isseus. It is recommended to not opt out
<code class="docutils literal notranslate"><span class="pre">drop_missing_pairs()</span></code> and <code class="docutils literal notranslate"><span class="pre">drop_broken_files()</span></code>. Otherwise, it will
meet exceptions.</p>
<p>If one wants filled hole, <code class="docutils literal notranslate"><span class="pre">fill_save_mask()</span></code> function will fill holes with
some tricks to handle edge cases and save them as .png format. Then set
<code class="docutils literal notranslate"><span class="pre">filled_mask</span></code> argument to True to load them.</p>
<p>Read more in Notes section</p>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul>
<li><p>4 channel PNG format annotation mask even though mask is binary. But it is
not grayscale binary. They put value 255 only in red channel.</p></li>
<li><p>Two annotation formats; Photoshop and GIMP. It seems that two annotators
worked separately. segmented-ashariff will be ignored. In total, 97
segmented images (out of 100)</p></li>
<li><dl>
<dt>3 missing segmentations: ind={31, 43, 75}</dt><dd><p>./data/images/dna-images/gnf/dna-31.png</p>
<p>./data/images/dna-images/gnf/dna-43.png</p>
<p>./data/images/dna-images/ic100/dna-25.png</p>
</dd>
</dl>
</li>
<li><p>Manually filled annotation to make masks using GIMP</p></li>
<li><p>2009_ISBI_2DNuclei_code_data/data/images/segmented-lpc/ic100/dna-15.xcf
does not have ‘borders’ layer like the others.  This one alone has
‘border’ layer.</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r392840f6ea36-1"><span class="brackets"><a class="fn-backref" href="#id51">1</a></span></dt>
<dd><p>L. P. Coelho, A. Shariff, and R. F. Murphy, “Nuclear segmentation in
microscope cell images: A hand-segmented dataset and comparison of
algorithms,” in 2009 IEEE International Symposium on Biomedical Imaging:
From Nano to Macro, Jun. 2009, pp. 518–521, doi:
10.1109/ISBI.2009.5193098.</p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.MurphyLab.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_murphylab.html#MurphyLab.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.MurphyLab.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.MurphyLab.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_murphylab.html#MurphyLab.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.MurphyLab.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.MurphyLab.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.MurphyLab.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.MurphyLab.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.MurphyLab.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.MurphyLab.fill_save_mask">
<span class="sig-name descname"><span class="pre">fill_save_mask</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_murphylab.html#MurphyLab.fill_save_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.MurphyLab.fill_save_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Fill holes from boundary mask with some tricks</p>
<p>Requires scipy and scikit-image. Install depencency with pip option
<code class="docutils literal notranslate"><span class="pre">pip</span> <span class="pre">install</span> <span class="pre">bioimageloader[process]</span></code>.</p>
<p>Note that this does not result perfect filled masks. Those not entirely
closed by this algorithm (36, 40, 63).</p>
<p>Other issues: <code class="docutils literal notranslate"><span class="pre">ind=63</span></code>: ‘border’ not ‘borders’, <code class="docutils literal notranslate"><span class="pre">ind=93</span></code>
<code class="docutils literal notranslate"><span class="pre">GimpDocument</span></code> cannot read it…</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.S_BSST265">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">S_BSST265</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_s_bsst265.html#S_BSST265"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.S_BSST265" title="Permalink to this definition">#</a></dt>
<dd><p>An annotated fluorescence image dataset for training nuclear segmentation
methods <a class="reference internal" href="#reb25356749b6-1" id="id55">[1]</a></p>
<p>Immuno Fluorescence (IF) images, designed for ML</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines
augmentation in sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>All images have grayscale BUT some have 3 channels</p></li>
<li><p>rawimages: Raw nuclear images in TIFF format</p></li>
<li><p>groundtruth: Annotated masks in TIFF format</p></li>
<li><p>groundtruth_svgs: SVG-Files for each annotated masks and corresponding raw
image in JPEG format</p></li>
<li><p>singlecell_groundtruth: Groundtruth for randomly selected nuclei of the
testset (25 nuclei per testset class, a subset of all nuclei of the
testset classes; human experts can compete with this low number of nuclei
per subset by calculating Dice coefficients between their annotations and
the groundtruth annotations)</p></li>
<li><p>visualized_groundtruth: Visualization of groundtruth masks in PNG format</p></li>
<li><p>visualized_singlecell_groundtruth: Visualization of groundtruth for
randomly selected nuclei in PNG format</p></li>
<li><p>Find more info in README.txt inside the root directory</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="reb25356749b6-1"><span class="brackets"><a class="fn-backref" href="#id55">1</a></span></dt>
<dd><p>F. Kromp et al., “An annotated fluorescence image dataset for
training nuclear segmentation methods,” Scientific Data, vol. 7, no. 1,
Art. no. 1, Aug. 2020, doi: 10.1038/s41597-020-00608-w.</p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.S_BSST265.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_s_bsst265.html#S_BSST265.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.S_BSST265.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.S_BSST265.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_s_bsst265.html#S_BSST265.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.S_BSST265.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.S_BSST265.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">list</span></em><a class="headerlink" href="#bioimageloader.collections.S_BSST265.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.S_BSST265.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.S_BSST265.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.StarDist">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">StarDist</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">training</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_stardist.html#StarDist"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.StarDist" title="Permalink to this definition">#</a></dt>
<dd><p>Dataset for StarDist <a class="reference internal" href="#r8947d9710bf3-1" id="id57">[1]</a>, <a class="reference internal" href="#r8947d9710bf3-2" id="id58">[2]</a></p>
<p>Cell Detection with Star-convex Polygons</p>
<p>StarDist data is a subset of Data Science Bowl 2018 <a class="reference internal" href="#r8947d9710bf3-3" id="id59">[3]</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines augmentation in
sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>training</strong><span class="classifier">bool, default: True</span></dt><dd><p>Load training set if True, else load testing one</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><a class="reference internal" href="#bioimageloader.collections.DSB2018" title="bioimageloader.collections.DSB2018"><code class="xref py py-obj docutils literal notranslate"><span class="pre">DSB2018</span></code></a></dt><dd></dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>StarDist data is a subset of Data Science Bowl 2018 <a class="reference internal" href="#r8947d9710bf3-3" id="id60">[3]</a>. Choose only one,
do not mix them.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">root_dir</span></code> is not ‘dsb2018’ even though the archive name is ‘dsb2018’,
because it conflicts with the original DSB2018. Make a new directory.</p></li>
<li><p>All images have grayscale</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r8947d9710bf3-1"><span class="brackets"><a class="fn-backref" href="#id57">1</a></span></dt>
<dd><p>U. Schmidt, M. Weigert, C. Broaddus, and G. Myers, “Cell Detection
with Star-convex Polygons,” arXiv:1806.03535 [cs], vol. 11071, pp.
265–273, 2018, doi: 10.1007/978-3-030-00934-2_30.</p>
</dd>
<dt class="label" id="r8947d9710bf3-2"><span class="brackets"><a class="fn-backref" href="#id58">2</a></span></dt>
<dd><p><a class="reference external" href="https://github.com/stardist/stardist/releases/download/0.1.0/dsb2018.zip">https://github.com/stardist/stardist/releases/download/0.1.0/dsb2018.zip</a></p>
</dd>
<dt class="label" id="r8947d9710bf3-3"><span class="brackets">3</span><span class="fn-backref">(<a href="#id59">1</a>,<a href="#id60">2</a>)</span></dt>
<dd><p><a class="reference external" href="https://www.kaggle.com/c/data-science-bowl-2018/">https://www.kaggle.com/c/data-science-bowl-2018/</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.StarDist.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_stardist.html#StarDist.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.StarDist.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.StarDist.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_stardist.html#StarDist.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.StarDist.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.StarDist.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.StarDist.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.StarDist.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.StarDist.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.TNBC">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">TNBC</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'cv2'</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_tnbc.html#TNBC"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.TNBC" title="Permalink to this definition">#</a></dt>
<dd><p>TNBC Nuclei Segmentation Dataset <a class="reference internal" href="#r7621ea10f8ba-1" id="id64">[1]</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’, ‘image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines
augmentation in sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘cv2’, ‘equal’, Sequence[float]}, default: ‘cv2’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along
channel axis, then divides it by the number of expected channels.</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r7621ea10f8ba-1"><span class="brackets"><a class="fn-backref" href="#id64">1</a></span></dt>
<dd><p>Segmentation of Nuclei in Histopathology Images by Deep Regression of
the Distance Map, <a class="reference external" href="https://ieeexplore.ieee.org/document/8438559">https://ieeexplore.ieee.org/document/8438559</a></p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.TNBC.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_tnbc.html#TNBC.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.TNBC.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.TNBC.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_tnbc.html#TNBC.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.TNBC.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.TNBC.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.TNBC.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.TNBC.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.TNBC.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>anno_dict[ind] = &lt;file&gt;</p>
</dd></dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="bioimageloader.collections.UCSB">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">bioimageloader.collections.</span></span><span class="sig-name descname"><span class="pre">UCSB</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">root_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'both'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transforms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">albumentations.core.composition.Compose</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">grayscale_mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'cv2'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">category</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">('malignant',)</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioimageloader/collections/_ucsb.html#UCSB"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.UCSB" title="Permalink to this definition">#</a></dt>
<dd><p>A biosegmentation benchmark for evaluation of bioimage analysis methods</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>root_dir</strong><span class="classifier">str</span></dt><dd><p>Path to root directory</p>
</dd>
<dt><strong>output</strong><span class="classifier">{‘both’,’ image’, ‘mask’}, default: ‘both’</span></dt><dd><p>Change outputs. ‘both’ returns {‘image’: image, ‘mask’: mask}.</p>
</dd>
<dt><strong>transforms</strong><span class="classifier">albumentations.Compose, optional</span></dt><dd><p>An instance of Compose (albumentations pkg) that defines
augmentation in sequence.</p>
</dd>
<dt><strong>num_samples</strong><span class="classifier">int, optional</span></dt><dd><p>Useful when <code class="docutils literal notranslate"><span class="pre">transforms</span></code> is set. Define the total length of the
dataset. If it is set, it overwrites <code class="docutils literal notranslate"><span class="pre">__len__</span></code>.</p>
</dd>
<dt><strong>grayscale</strong><span class="classifier">bool, default: False</span></dt><dd><p>Convert images to grayscale</p>
</dd>
<dt><strong>grayscale_mode</strong><span class="classifier">{‘cv2’, ‘equal’, Sequence[float]}, default: ‘cv2’</span></dt><dd><p>How to convert to grayscale. If set to ‘cv2’, it follows opencv
implementation. Else if set to ‘equal’, it sums up values along
channel axis, then divides it by the number of expected channels.</p>
</dd>
<dt><strong>category</strong><span class="classifier">{‘benign’, ‘malignant’}, default: (‘malignant’,)</span></dt><dd><p>Select which category of output you want</p>
</dd>
</dl>
</dd>
</dl>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">MaskDataset</span></code></dt><dd><p>Super class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">Dataset</span></code></dt><dd><p>Base class</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">DatasetInterface</span></code></dt><dd><p>Interface</p>
</dd>
</dl>
</div>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>32 ‘benign’, 26 ‘malignant’ images (58 images in total)</p></li>
<li><p>58x768x896 -&gt; ~600 patches. Thus, the defulat <cite>num_samples=900</cite> (x1.5).</p></li>
<li><p>Images are not fully annotated</p></li>
</ul>
<p class="rubric">References</p>
<dl class="citation">
<dt class="label" id="r0c096f2eda07-1"><span class="brackets">1</span></dt>
<dd><p>E. Drelie Gelasca, B. Obara, D. Fedorov, K. Kvilekval, and B.
Manjunath, “A biosegmentation benchmark for evaluation of bioimage
analysis methods,” BMC Bioinformatics, vol. 10, p. 368, Nov. 2009, doi:
10.1186/1471-2105-10-368.</p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.UCSB.get_image">
<span class="sig-name descname"><span class="pre">get_image</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_ucsb.html#UCSB.get_image"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.UCSB.get_image" title="Permalink to this definition">#</a></dt>
<dd><p>Get an image</p>
</dd></dl>
<dl class="py method">
<dt class="sig sig-object py" id="bioimageloader.collections.UCSB.get_mask">
<span class="sig-name descname"><span class="pre">get_mask</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">pathlib.Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.ndarray</span></span></span><a class="reference internal" href="../_modules/bioimageloader/collections/_ucsb.html#UCSB.get_mask"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#bioimageloader.collections.UCSB.get_mask" title="Permalink to this definition">#</a></dt>
<dd><p>Get a mask</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.UCSB.file_list">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">file_list</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.UCSB.file_list" title="Permalink to this definition">#</a></dt>
<dd><p>A list of pathes to image files</p>
</dd></dl>
<dl class="py property">
<dt class="sig sig-object py" id="bioimageloader.collections.UCSB.anno_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">anno_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">int</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">pathlib.Path</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#bioimageloader.collections.UCSB.anno_dict" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of pathes to annotation files</p>
</dd></dl>
</dd></dl>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="batch.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">bioimageloader.batch</div>
              </div>
              <svg><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="common.html">
              <svg><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">bioimageloader.common</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2022, Seongbin Lim
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            <div class="icons">
              <a class="muted-link " href="https://github.com/LaboratoryOpticsBiosciences/bioimageloader" aria-label="GitHub">
                <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 16 16">
                    <path fill-rule="evenodd" d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"></path>
                </svg>
            </a>
              
            </div>
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer no-toc">
      
      
      
    </aside>
  </div>
</div><script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/scripts/furo.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    </body>
</html>